reconcile/external_resources/aws.py

```diff
@@ -142,9 +175,53 @@ def resolve(self, spec: ExternalResourceSpec) -> dict[str, Any]:
                 spec.provisioner_name, kms_key_id
             ).identifier
 
+        # If not timeouts are set, set default timeouts according to the module reconcile timeout configuration
+        # 5 minutes are substracted to let terraform finish gracefully before the Job is killed.
+        if "timeouts" not in data:
+            data["timeouts"] = {
+                "create": f"{module_conf.reconcile_timeout_minutes - 5}m",
+                "update": f"{module_conf.reconcile_timeout_minutes - 5}m",
+                "delete": f"{module_conf.reconcile_timeout_minutes - 5}m",
+            }
         return data
 
-    def validate(self, resource: ExternalResource) -> None: ...
+    def _get_timeout_value(
+        self,
+        timeout: str,
+    ) -> int:
+        if not re.match(AWSRdsFactory.TIMEOUT_RE, timeout):
+            raise ValueError(
+                f"Invalid RDS instance timeout format: {timeout}. Specify timeout in minutes(m) or hours(h)."
+            )
+        value, unit = timeout[:-1], timeout[-1]
+        return int(value) * AWSRdsFactory.TIMEOUT_UNIT_CONVERSION[unit]
+
+    def _validate_timeouts(
+        self,
+        resource: ExternalResource,
+        module_conf: ExternalResourceModuleConfiguration,
+    ) -> None:
+        timeouts = resource.data.get("timeouts")
+        if not timeouts:
+            return
+
+        for option in ("create", "update", "delete"):
+            timeout = timeouts.get(option)
+            if (
+                timeout
+                and (value := self._get_timeout_value(timeout))
+                > module_conf.reconcile_timeout_minutes
+            ):
+                raise ValueError(
+                    f"RDS instance {option} timeout value {value} exceeds the module reconcile_timeout_minutes value {module_conf.reconcile_timeout_minutes}."
+                )
```

Comment:
also need to ensure `timeouts` to be a `dict` with only allowed keys (`create`, `update`, `delete`), either use a new BaseModel or manually verify like

```python
        if not isinstance(timeouts, dict):
            raise ValueError()

        allowed_keys = {"create", "update", "delete"}
        if unknown_keys := timeouts.keys() - allowed_keys:
            raise ValueError()

        for option, timeout in timeouts.items():
            timeout_minutes = self._get_timeout_minutes(timeout)
            if timeout_minutes > module_conf.reconcile_timeout_minutes:
                raise ValueError()
```

Reply:
Done.